iothub/device/src/Transport/Http/HttpClientHelper.cs

```diff
@@ -123,11 +118,11 @@ internal sealed class HttpClientHelper : IHttpClientHelper
                 _httpClientHandler.UseProxy = proxy != null;
                 _httpClientHandler.Proxy = proxy;
             }
+#endif
 
-            _httpClientHandler.MaxConnectionsPerServer = DefaultMaxConnectionsPerServer;
+            ServicePointHelpers.SetLimits(_httpClientHandler, _baseAddress);
 
-            _httpClientObj = new HttpClient(_httpClientHandler);
-#endif
+            _httpClientObj = _httpClientHandler != null ? new HttpClient(_httpClientHandler) : new HttpClient();
```

Comment:
_httpClientHandler can be null, but the call to SetLimits doesn't seem to handle that well.

Reply:
I don't think it can be null, per other code above.

Reply:
I'll make the service point helpers method handle null cases here just in case